### PR TITLE
(BKR-275) PowerShell Wrapper Does not Handle Quoting

### PR DESCRIPTION
**Pull request by anodelman**
- create new powershell dsl helper:
  execute_powershell_script
  * takes as input a string representing a powershell script
  * create a file on the host containing the script
  * executes the script using powershell -File

### CODE REVIEW

File: lib/beaker/dsl/wrappers.rb [hunk 1]
```diff
@@ -124,19 +131,47 @@ def powershell(command, args={})
...
+
+      # Convert the provided command string to Base64
+      # @param [String] cmd The command to convert to Base64
+      # @return [String] The converted string
+      # @api private
+      def encode_command(cmd)
+        cmd = cmd.chars.to_a.join("\x00").chomp
+        cmd << "\x00" unless cmd[-1].eql? "\x00"
+        if(defined?(cmd.encode))
+          cmd = cmd.encode('ASCII-8BIT')
```
Comment (cowofevil) [inline lib/beaker/dsl/wrappers.rb]: Why not support Unicode?

**Comment by anodelman** [comment]
Failure unrelated to patch.  (/me shakes fist at centos7)

File: lib/beaker/dsl/wrappers.rb [hunk 2]
```diff
@@ -124,19 +131,43 @@ def powershell(command, args={})
           'NoProfile'       => '',
           'NonInteractive'  => ''
         }
+        encoded = false
         ps_opts.merge!(args)
         ps_args = []
         ps_opts.each do |k, v|
           if v.eql?('') or v.nil?
             ps_args << "-#{k}"
+          elsif k.eql?('EncodedCommand') && v
```
Comment (colinPL) [inline lib/beaker/dsl/wrappers.rb]: If `k = EncodedCommand` and `v = nil`, then the result will be `ps_args = [ ..., -EncodedCommand]`. 

Is that command argument valid?

**Comment by colinPL** [comment]
:+1:

**Comment by kevpl** [comment]
talked with @anodelman, additional spec testing to come later today.

**Comment by anodelman** [comment]
Added spec coverage.

**Comment by anodelman** [comment]
Failure unrelated to patch.

**Pull request merged**